### PR TITLE
Feature/server/map display/api fetch locations

### LINKED ISSUE #13: Server/Map-Display: Create API route for fetching locations

### PR DESCRIPTION
**Pull request by perlaescano**
## Description
- Added a new API endpoint `/locations` to fetch all_location data from the MongoDB database.
- Created the `Location` model in `server/models/location.model.js` to define the schema for locations.
- Implemented the location route in `server/routes/location.route.js` to handle API requests.
- Updated the `Home` component in `Home.jsx` to fetch location data.
- Corrected some spelling mistakes in the user route (`server/routes/user.route.js`).

## What's in this change?
- API endpoint to fetch location data.
- New Mongoose model for handling location data (`Location`).
- Backend route for fetching locations from the `all_locations` collection.
- Integration with the frontend to fetch and display location data in the `Home` component.

## Testing
- Tested the `/locations` endpoint using Postman to verify data retrieval from MongoDB.
- Verified that the frontend correctly fetches and displays location data in the console.
- Checked the integration between the frontend and backend locally.
- Deployed committed changes to Render and verified functionality on the live environment.

### CODE REVIEW

File: server/routes/location.route.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import express from 'express';
+import Location from '../models/location.model.js';  
+
+const router = express.Router();
+
+router.get('/', async (req, res) => {
```
Comment (ghost) [inline server/routes/location.route.js]: Please make a get request on route /locations

File: client/src/pages/Home.jsx [hunk 2]
```diff
@@ -1,25 +1,36 @@
-import React, {useState, useEffect} from 'react';
+import React, { useState, useEffect } from 'react';
 import NavBar from '../components/NavBar.jsx';
-import '../style/Home.css';
```
Comment (ghost) [inline client/src/pages/Home.jsx:3]: Add back the styling sheet instead delete the unnecessary css class indicator.

**Review by ghost — CHANGES_REQUESTED** [review]
Good work! Have a look at my two comments!

File: server/models/location.model.js
Comment (keunsublee) [inline server/models/location.model.js:1]: will need address and multipolygon if we want to use data with no lat/lon (Sports, Athletic Facilities), we could drop these though

**Review by keunsublee — APPROVED** [review]

File: server/models/location.model.js
Comment (perlaescano) [reply server/models/location.model.js:1]: yes I will address that in another task/branch setup (Server/Map-Display: Setup geospatial queries)

**Review by ghost — APPROVED** [review]
Good JOB!

**Pull request merged**